### PR TITLE
Windup 597 - Extract hibernate information to type server resources

### PR DESCRIPTION
**Pull request by bradsdavis**

### CODE REVIEW

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/764/<h2>Failed Tests: <span class='status-failure'>4</span></h2><h3><a name='org.jboss.windup:windup-tooling-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/764/org.jboss.windup$windup-tooling-tests/testReport'>org.jboss.windup:windup-tooling-tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/764/org.jboss.windup$windup-tooling-tests/testReport/org.jboss.windup.tooling/ExecutionBuilderTest/testExecutionBuilder/'><strong>org.jboss.windup.tooling.ExecutionBuilderTest.testExecutionBuilder</strong></a></li></ul><h3><a name='org.jboss.windup.tests:windup-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/764/org.jboss.windup.tests$windup-tests/testReport'>org.jboss.windup.tests:windup-tests</a>: <span class='status-failure'>3</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/764/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureSpringSmallTest/testRunWindupSmallSpringApp/'><strong>org.jboss.windup.tests.application.WindupArchitectureSpringSmallTest.testRunWindupSmallSpringApp</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/764/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureJEEExampleTest/testRunWindupJEEExampleMode/'><strong>org.jboss.windup.tests.application.WindupArchitectureJEEExampleTest.testRunWindupJEEExampleMode</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/764/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureSourceModeTest/testRunWindupSourceMode/'><strong>org.jboss.windup.tests.application.WindupArchitectureSourceModeTest.testRunWindupSourceMode</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/133/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='org.jboss.windup.rules:windup-rulesets' /><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/133/org.jboss.windup.rules$windup-rulesets/testReport'>org.jboss.windup.rules:windup-rulesets</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/133/org.jboss.windup.rules$windup-rulesets/testReport/org.jboss.windup.rules.tests/WindupRulesTest/testWindupRules/'><strong>org.jboss.windup.rules.tests.WindupRulesTest.testWindupRules</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/765/<h2>Failed Tests: <span class='status-failure'>4</span></h2><h3><a name='org.jboss.windup:windup-tooling-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/765/org.jboss.windup$windup-tooling-tests/testReport'>org.jboss.windup:windup-tooling-tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/765/org.jboss.windup$windup-tooling-tests/testReport/org.jboss.windup.tooling/ExecutionBuilderTest/testExecutionBuilder/'><strong>org.jboss.windup.tooling.ExecutionBuilderTest.testExecutionBuilder</strong></a></li></ul><h3><a name='org.jboss.windup.tests:windup-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/765/org.jboss.windup.tests$windup-tests/testReport'>org.jboss.windup.tests:windup-tests</a>: <span class='status-failure'>3</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/765/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureSpringSmallTest/testRunWindupSmallSpringApp/'><strong>org.jboss.windup.tests.application.WindupArchitectureSpringSmallTest.testRunWindupSmallSpringApp</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/765/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureJEEExampleTest/testRunWindupJEEExampleMode/'><strong>org.jboss.windup.tests.application.WindupArchitectureJEEExampleTest.testRunWindupJEEExampleMode</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/765/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureSourceModeTest/testRunWindupSourceMode/'><strong>org.jboss.windup.tests.application.WindupArchitectureSourceModeTest.testRunWindupSourceMode</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/134/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='org.jboss.windup.rules:windup-rulesets' /><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/134/org.jboss.windup.rules$windup-rulesets/testReport'>org.jboss.windup.rules:windup-rulesets</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/134/org.jboss.windup.rules$windup-rulesets/testReport/org.jboss.windup.rules.tests/WindupRulesTest/testWindupRules/'><strong>org.jboss.windup.rules.tests.WindupRulesTest.testWindupRules</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/766/<h2>Failed Tests: <span class='status-failure'>4</span></h2><h3><a name='org.jboss.windup:windup-tooling-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/766/org.jboss.windup$windup-tooling-tests/testReport'>org.jboss.windup:windup-tooling-tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/766/org.jboss.windup$windup-tooling-tests/testReport/org.jboss.windup.tooling/ExecutionBuilderTest/testExecutionBuilder/'><strong>org.jboss.windup.tooling.ExecutionBuilderTest.testExecutionBuilder</strong></a></li></ul><h3><a name='org.jboss.windup.tests:windup-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/766/org.jboss.windup.tests$windup-tests/testReport'>org.jboss.windup.tests:windup-tests</a>: <span class='status-failure'>3</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/766/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureSpringSmallTest/testRunWindupSmallSpringApp/'><strong>org.jboss.windup.tests.application.WindupArchitectureSpringSmallTest.testRunWindupSmallSpringApp</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/766/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureJEEExampleTest/testRunWindupJEEExampleMode/'><strong>org.jboss.windup.tests.application.WindupArchitectureJEEExampleTest.testRunWindupJEEExampleMode</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/766/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureSourceModeTest/testRunWindupSourceMode/'><strong>org.jboss.windup.tests.application.WindupArchitectureSourceModeTest.testRunWindupSourceMode</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/135/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='org.jboss.windup.rules:windup-rulesets' /><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/135/org.jboss.windup.rules$windup-rulesets/testReport'>org.jboss.windup.rules:windup-rulesets</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/135/org.jboss.windup.rules$windup-rulesets/testReport/org.jboss.windup.rules.tests/WindupRulesTest/testWindupRules/'><strong>org.jboss.windup.rules.tests.WindupRulesTest.testWindupRules</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/767/<h2>Failed Tests: <span class='status-failure'>4</span></h2><h3><a name='org.jboss.windup:windup-tooling-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/767/org.jboss.windup$windup-tooling-tests/testReport'>org.jboss.windup:windup-tooling-tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/767/org.jboss.windup$windup-tooling-tests/testReport/org.jboss.windup.tooling/ExecutionBuilderTest/testExecutionBuilder/'><strong>org.jboss.windup.tooling.ExecutionBuilderTest.testExecutionBuilder</strong></a></li></ul><h3><a name='org.jboss.windup.tests:windup-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/767/org.jboss.windup.tests$windup-tests/testReport'>org.jboss.windup.tests:windup-tests</a>: <span class='status-failure'>3</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/767/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureSpringSmallTest/testRunWindupSmallSpringApp/'><strong>org.jboss.windup.tests.application.WindupArchitectureSpringSmallTest.testRunWindupSmallSpringApp</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/767/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureJEEExampleTest/testRunWindupJEEExampleMode/'><strong>org.jboss.windup.tests.application.WindupArchitectureJEEExampleTest.testRunWindupJEEExampleMode</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/767/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureSourceModeTest/testRunWindupSourceMode/'><strong>org.jboss.windup.tests.application.WindupArchitectureSourceModeTest.testRunWindupSourceMode</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/136/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='org.jboss.windup.rules:windup-rulesets' /><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/136/org.jboss.windup.rules$windup-rulesets/testReport'>org.jboss.windup.rules:windup-rulesets</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/136/org.jboss.windup.rules$windup-rulesets/testReport/org.jboss.windup.rules.tests/WindupRulesTest/testWindupRules/'><strong>org.jboss.windup.rules.tests.WindupRulesTest.testWindupRules</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/773/<h2>Failed Tests: <span class='status-failure'>4</span></h2><h3><a name='org.jboss.windup:windup-tooling-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/773/org.jboss.windup$windup-tooling-tests/testReport'>org.jboss.windup:windup-tooling-tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/773/org.jboss.windup$windup-tooling-tests/testReport/org.jboss.windup.tooling/ExecutionBuilderTest/testExecutionBuilder/'><strong>org.jboss.windup.tooling.ExecutionBuilderTest.testExecutionBuilder</strong></a></li></ul><h3><a name='org.jboss.windup.tests:windup-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/773/org.jboss.windup.tests$windup-tests/testReport'>org.jboss.windup.tests:windup-tests</a>: <span class='status-failure'>3</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/773/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureSpringSmallTest/testRunWindupSmallSpringApp/'><strong>org.jboss.windup.tests.application.WindupArchitectureSpringSmallTest.testRunWindupSmallSpringApp</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/773/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureJEEExampleTest/testRunWindupJEEExampleMode/'><strong>org.jboss.windup.tests.application.WindupArchitectureJEEExampleTest.testRunWindupJEEExampleMode</strong></a></li><li><a href='http://10.3.8.5/job/windup-pr-builder/773/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureSourceModeTest/testRunWindupSourceMode/'><strong>org.jboss.windup.tests.application.WindupArchitectureSourceModeTest.testRunWindupSourceMode</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/142/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='org.jboss.windup.rules:windup-rulesets' /><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/142/org.jboss.windup.rules$windup-rulesets/testReport'>org.jboss.windup.rules:windup-rulesets</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/142/org.jboss.windup.rules$windup-rulesets/testReport/org.jboss.windup.rules.tests/WindupRulesTest/testWindupRules/'><strong>org.jboss.windup.rules.tests.WindupRulesTest.testWindupRules</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/774/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='org.jboss.windup.tests:windup-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/774/org.jboss.windup.tests$windup-tests/testReport'>org.jboss.windup.tests:windup-tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/774/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureSpringSmallTest/testRunWindupSmallSpringApp/'><strong>org.jboss.windup.tests.application.WindupArchitectureSpringSmallTest.testRunWindupSmallSpringApp</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/143/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='org.jboss.windup.rules:windup-rulesets' /><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/143/org.jboss.windup.rules$windup-rulesets/testReport'>org.jboss.windup.rules:windup-rulesets</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/143/org.jboss.windup.rules$windup-rulesets/testReport/org.jboss.windup.rules.tests/WindupRulesTest/testWindupRules/'><strong>org.jboss.windup.rules.tests.WindupRulesTest.testWindupRules</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/775/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='org.jboss.windup.tests:windup-tests' /><a href='http://10.3.8.5/job/windup-pr-builder/775/org.jboss.windup.tests$windup-tests/testReport'>org.jboss.windup.tests:windup-tests</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-pr-builder/775/org.jboss.windup.tests$windup-tests/testReport/org.jboss.windup.tests.application/WindupArchitectureSpringSmallTest/testRunWindupSmallSpringApp/'><strong>org.jboss.windup.tests.application.WindupArchitectureSpringSmallTest.testRunWindupSmallSpringApp</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/144/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='org.jboss.windup.rules:windup-rulesets' /><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/144/org.jboss.windup.rules$windup-rulesets/testReport'>org.jboss.windup.rules:windup-rulesets</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/144/org.jboss.windup.rules$windup-rulesets/testReport/org.jboss.windup.rules.tests/WindupRulesTest/testWindupRules/'><strong>org.jboss.windup.rules.tests.WindupRulesTest.testWindupRules</strong></a></li></ul>
Test FAILed.

File: reporting/api/src/main/java/org/jboss/windup/reporting/model/association/LinkableModel.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+import org.jboss.windup.graph.model.WindupVertexFrame;
+import org.jboss.windup.reporting.model.LinkModel;
+
+import com.tinkerpop.blueprints.Direction;
+import com.tinkerpop.frames.Adjacency;
+import com.tinkerpop.frames.modules.typedgraph.TypeValue;
+
+@TypeValue(LinkableModel.TYPE)
+public interface LinkableModel extends WindupVertexFrame
```
Comment (lincolnthree) [inline reporting/api/src/main/java/org/jboss/windup/reporting/model/association/LinkableModel.java]: In the future, try to add a little type javadoc about the general concept of what a model is for. Helps us keep it all straight :) Good job on the method api docs though those could use a little more explanation, too.  thanks.

**Pull request merged**